src/traces/parcoords/defaults.js

```diff
@@ -51,29 +51,41 @@ function dimensionsDefaults(traceIn, traceOut) {
             continue;
         }
 
-        coerce('label');
-        coerce('tickvals');
-        coerce('ticktext');
-        coerce('visible');
-        coerce('range');
-        coerce('constraintrange');
-        coerce('values');
+        var userVisible = coerce('visible');
 
         // turn dimensions with no data invisible
-        dimensionOut.visible = dimensionOut.visible && dimensionOut.values.length > 0;
+        var actuallyVisible = userVisible && Lib.isArray(dimensionIn.values) && dimensionIn.values.length > 0;
+
+        if(actuallyVisible) {
```

Comment:
Commit https://github.com/plotly/plotly.js/commit/c6b8d3441eb572191b93b25339a6151a7c3552d3 shows what I was hoping for in https://github.com/plotly/plotly.js/pull/1256#discussion_r99377349

Does this break anything downstream?